ts/features/bonus/cgn/store/reducers/eyca/activation.ts

```diff
@@ -0,0 +1,73 @@
+// bonus reducer
+import { getType } from "typesafe-actions";
+import { createSelector } from "reselect";
+import { Action } from "../../../../../../store/actions/types";
+import { GlobalState } from "../../../../../../store/reducers/types";
+import {
+  cgnEycaActivationRequest,
+  cgnEycaActivationStatus
+} from "../../actions/eyca/activation";
+import { EycaActivationDetail } from "../../../../../../../definitions/cgn/EycaActivationDetail";
+
+export enum CgnEycaActivationProgressEnum {
+  "UNDEFINED" = "UNDEFINED",
+  "TIMEOUT" = "TIMEOUT", // number of polling exceeded
+  "PROGRESS" = "PROGRESS", // The request is started
+  "PENDING" = "PENDING", // Polling time exceeded
+  "ERROR" = "ERROR", // There's an error
+  "EXISTS" = "EXISTS", // Another bonus related to this user was found
+  "INELIGIBLE" = "INELIGIBLE", // Another bonus related to this user was found
+  "SUCCESS" = "SUCCESS" // Activation has been completed
+}
+
+export type EycaActivationState = {
+  status: CgnEycaActivationProgressEnum;
+  value?: EycaActivationDetail;
+};
+
+const INITIAL_STATE: EycaActivationState = {
+  status: CgnEycaActivationProgressEnum.UNDEFINED
+};
+
+const reducer = (
+  state: EycaActivationState = INITIAL_STATE,
+  action: Action
+): EycaActivationState => {
+  switch (action.type) {
+    // bonus activation
+    case getType(cgnEycaActivationRequest):
+    case getType(cgnEycaActivationStatus.request):
+      return {
+        ...state,
+        status: CgnEycaActivationProgressEnum.PROGRESS
+      };
+    case getType(cgnEycaActivationStatus.success):
+      return {
+        ...state,
+        status: action.payload.status,
+        value: action.payload.value
+      };
+    case getType(cgnEycaActivationStatus.failure):
+      return {
+        ...state,
+        status: CgnEycaActivationProgressEnum.ERROR
+      };
+  }
+  return state;
+};
+
+// Selectors
+export const eycaActivationSelector = (
+  state: GlobalState
+): EycaActivationState => state.bonus.cgn.eyca.activation;
+
+export const isCgnActivationLoading = createSelector<
+  GlobalState,
+  EycaActivationState,
+  boolean
+>(
+  eycaActivationSelector,
+  activation => activation.status !== CgnEycaActivationProgressEnum.ERROR
```

Comment:
```suggestion
  activation => activation.status === CgnEycaActivationProgressEnum.PROGRESS
```
Why don't you use the specific status?

Reply:
This selector will only be used in an error/loading screen to check what to show between the error or loading component

going to add a comment what do you think?

Reply:
Instead of adding a comment could be a better option to expose the information as it is without any logic?
It seems you are putting an UI login inside the selector: if I use (as a devloper) `isCgnActivationLoading` I wish have `true` when it is in loading state, not true when it is not in error.

So my suggestion is:  keep the selector "_pure_" and move that logic locally into the screen

```typescript
// return the cgn eyca status
export const cgnEycaActivationstatus = createSelector(
  eycaActivationSelector,
  (activation: EycaActivationState): CgnEycaActivationProgressEnum =>
    activation.status
);
```

and use it in the Screen as follows

```typescript
const mapStateToProps = (state: GlobalState) => ({
  isCgnEycaStatusInError:
    cgnEycaActivationstatus(state) === CgnEycaActivationProgressEnum.ERROR
});
```